lib/tasks/populate_canonical_models.rake

```diff
@@ -89,208 +119,270 @@ namespace :canonical_models do
+          Rails.logger.error "Unknown error saving canonical material \"#{canonical_material_attributes[:item_code]}\": #{e.message}"
           raise e
         end
       end
     end
 
     desc 'Populate or update canonical jewelry items from JSON data'
-    task canonical_jewelry: :environment do
-      Rails.logger.info 'Populating canonical jewelry items...'
-
-      items = JSON.parse(File.read(Rails.root.join('lib', 'tasks', 'canonical_models', 'canonical_jewelry.json')), symbolize_names: true)
-
-      ActiveRecord::Base.transaction do
-        items.each do |data|
-          item = CanonicalJewelryItem.find_or_initialize_by(name: data[:attributes][:name])
-          item.assign_attributes(data[:attributes])
-          item.save!
-
-          if data.has_key?(:materials)
-            data[:materials].each do |m|
-              material = CanonicalMaterial.find_by(name: m[:name])
-
-              if material.present? && item.canonical_materials.exclude?(material)
-                CanonicalJewelryItemsCanonicalMaterial.create!(
-                  canonical_jewelry_item: item,
-                  canonical_material:     material,
-                  quantity:               m[:quantity],
-                )
-              elsif item.canonical_materials.include?(material)
-                Rails.logger.warn("Jewelry item #{item.item_code} already associated with material #{m[:name]}.")
-              else
-                Rails.logger.warn("Jewelry item #{item.item_code} calls for material #{m[:name]} but material does not exist.")
-              end
-            end
-          end
-
-          if data.has_key?(:enchantments)
-            data[:enchantments].each do |en|
-              enchantment = Enchantment.find_by(name: en[:name])
-
-              if enchantment.present? && item.enchantments.exclude?(enchantment)
-                CanonicalJewelryItemsEnchantment.create!(
-                  canonical_jewelry_item: item,
-                  enchantment:            enchantment,
-                  strength:               en[:strength],
-                )
-              elsif item.enchantments.include?(enchantment)
-                Rails.logger.warn("Jewelry item #{item.item_code} already associated with enchantment #{enchantment.name}.")
-              else
-                Rails.logger.warn("Jewelry item #{item.item_code} calls for enchantment #{en[:name]} but enchantment does not exist.")
-              end
-            end
-          end
-        rescue ActiveRecord::RecordInvalid => e
-          Rails.logger.error "Validation error saving canonical jewelry item \"#{data[:attributes][:item_code]}\": #{e.message}"
-          raise e
-        rescue StandardError => e
-          Rails.logger.error "Unknown error saving canonical jewelry item \"#{data[:attributes][:item_code]}\": #{e.message}"
-          raise e
-        end
-      end
-    end
+    task :canonical_jewelry,
```

Comment:
I was very conflicted about this line and the other lines like it. I feel it would've read better had I kept it all on one line, but that line was simply too long. I found it would be less readable if the code was all on one line, despite the multi-line way not being the best either.